### DISCUSSION TITLE
Teensy 4.1/T41U5XBB - Plasma THC...

### DISCUSSION THREAD

**Discussion by sntlewis** [grblHAL/core#345] [Q&A]
Ok a few disclaimers...
1) rank newby to GRBL/GRBLHal world..
2) very familiar with Marlin/MACH3 (built several 3d printers and cnc routers)
3) very early in my next "build."  This will be a CNC Plasma table.  I already have the design/build underway albeit slow due to $$$.
4) understand that GRBLHal Plasma is young and under development.

Now for the question of the day as I dig into/learn GRBLHal.  Other than modifying my.machine.h to enable "plasma" what else is needed to be done before compile/flash.  My MCU is the Teensy 4.1 with the T41U5XBB BOB.

**Comment by terjeio** [comment]
If you want to add THC support analog input is required. @phil-barrett has a daugtherboard for that.

**Reply by sntlewis** [reply]
thanks, I'll reach out to him.

**Reply by sntlewis** [reply]
I'm still waiting for Phil to msg back..  But if I set PLASMA_ENABLE to 1 in my_machine.h and compile/build/flash...

I should see a panel of THC options like I do when setting ETHERNET_ENABLE under IOSender.  Correct?

**Reply by terjeio** [reply]
This is the status panel made way back:

![image](https://github.com/grblHAL/core/assets/20260062/5b60e940-a4ea-4b71-9a6e-b20845d8c54b)

It requires a new edge build of the sender + an instantiated plasma module in the controller to become available. I'll report back when I have made them available.

**Reply by terjeio** [reply]
Oops, if you meant in the _Settings: Grbl_ tab then yes, they become available under a new group _Plasma_. Binding inputs to aux ports can be configured under the _Aux ports_ group.

**Reply by sntlewis** [reply]
Ok, I don't see either so Ill wait for the next sender and plasma module...
